frontend/pages/hosts/ManageHostsPage/components/SecretEditorModal/SecretEditorModal.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { useState, useCallback } from "react";
+import { useSelector } from "react-redux";
+import Modal from "components/Modal";
+import Button from "components/buttons/Button";
+// @ts-ignore
+import InputField from "components/forms/fields/InputField";
+import { ITeam } from "interfaces/team";
+import { IEnrollSecret } from "interfaces/enroll_secret";
+
+interface IAddSecretModal {
+  selectedTeam: number;
+  onSaveSecret: (newEnrollSecret: string) => void;
+  teams: ITeam[];
+  toggleSecretEditorModal: () => void;
+  selectedSecret: IEnrollSecret | undefined;
+}
+
+interface IRootState {
+  app: {
+    enrollSecret: IEnrollSecret[];
+  };
+}
+
+const baseClass = "secret-editor-modal";
+
+const randomSecretGenerator = () => {
+  const randomChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz";
+  let result = "";
+  for (let i = 0; i < 32; i += 1) {
+    result += randomChars.charAt(
+      Math.floor(Math.random() * randomChars.length)
+    );
+  }
+  return result;
+};
```

Comment:
This is fine where it's at, but good to keep in mind for possibly moving to a global utility at some point if we find ourselves needing to generate secrets elsewhere.